packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -46,20 +46,34 @@ import { pathComposer } from '../3rdparty/polymer/path-composer';
 import { getInternalChildNodes, setNodeKey, setNodeOwnerKey } from './node';
 import { innerHTMLSetter } from '../env/element';
 import { getOwnerDocument } from '../shared/utils';
+import { adoptHostToSupportSlotting } from './slot';
 
 const { getHiddenField, setHiddenField, createFieldName } = fields;
 const ShadowRootResolverKey = '$shadowResolver$';
 const InternalSlot = createFieldName('shadowRecord', 'synthetic-shadow');
 const { createDocumentFragment } = document;
 
+export interface SlottedNodeRecord {
+    // slotted node
+    0: Node;
+    // slot name
+    1: string;
+}
+
 interface ShadowRootRecord {
     mode: 'open' | 'closed';
     delegatesFocus: boolean;
     host: Element;
     shadowRoot: SyntheticShadowRootInterface;
+    // Record with all slot elements ever inserted in the shadowRoot
+    slotElements: Record<string, HTMLSlotElement>;
+    // Array of records with all slotted nodes in the order of insertion
+    slottedRecords: SlottedNodeRecord[];
 }
 
-function getInternalSlot(root: SyntheticShadowRootInterface | Element): ShadowRootRecord {
+export function getShadowRootRecord(
```

Comment:
renaming and exporting... since we want to use the record from other files as well.